### ISSUE TITLE
typescript: defineProps and defineEmits not co-op in type checking

### ISSUE THREAD

**Issue by TrungRueta** [antfu/unplugin-vue2-script-setup#31]
Hi!

I'm using plugin v0.5.4 in Nuxt. with simple component use both `defineProps` and `defineEmits` i see type checking report error:

normal, when only use `defineProps` component checking ok:

![without defineEmits](https://user-images.githubusercontent.com/12226007/131468178-8d481eca-2798-4115-b6dd-2a343b5442df.png)

when add `defineEmis`, all props reference in template lost:

![with defineEmits](https://user-images.githubusercontent.com/12226007/131468222-5b17fa30-a28c-436b-834d-4b3edcdede64.png)

This error only happend with type checking, runtime still work.

Thankyou!

**Comment by antfu** [comment]
Have you configured this? https://github.com/antfu/unplugin-vue2-script-setup#global-types

**Comment by TrungRueta** [comment]
@antfu yes, this is in my tsconfig.json:

```
"types": [
      "@nuxt/types",
      "@types/node",
      "@nuxt/image",
      "@nuxtjs/firebase",
      "unplugin-vue2-script-setup/types",
      "unplugin-vue2-script-setup/ref-macros"
    ]
```

also, i found in readme we missing a thing about different between vue2's template and vue3's template type checking. i will post other issue for track better.

**Comment by antfu** [comment]
Fixed in @vue/composition-api 1.1.4, please update and try out
